### ISSUE TITLE
Redirect to https

### ISSUE THREAD

**Issue by Munter** [expressjs/expressjs.com#955]
The current documentation responds to `http:` requests. It would improve security if it would redirect to the `https:` equivalent instead

**Comment by crandmck** [comment]
So GitHub started supporting this in May (cf. https://blog.github.com/2018-05-01-github-pages-custom-domains-https/).  

But it looks like we can't do this until we update the DNS record (per https://help.github.com/articles/troubleshooting-custom-domains/#https-errors).  However, IBM still controls this AFAIK.  

@rmg Can you make this change?

**Comment by dougwilson** [comment]
Yea. The dns is currently a cloudflare account. We can enable https only through there too. A while back there concern about https only though cloudflare was that if we needed to go back to github direct it would break https links.

Now that github directly supports https there is no concern, so even just flipping on https only in cloudflare is doable, whichever is easier.

**Comment by annevk** [comment]
Any update on this?

**Comment by dougwilson** [comment]
Rand, do you know who has access to the cloudflare account? I think it has all been transferred to node.js foundation last I remember.

**Comment by Knighton910** [comment]
Im glad you guys are talking about this issue, If you need any help let me know.

**Comment by crandmck** [comment]
This was discussed some time ago: https://github.com/expressjs/discussions/issues/68#issuecomment-421428179 but I think someone in Node Foundation needs to do something....  Maybe we can get their attention? @mikeal

**Comment by Knighton910** [comment]
Why are you even here then Rand (haha). Just Kidding, i am grateful for all you did do for the express Project. Anyway hopefully we will get it sorted out soon.

**Comment by ryhinchey** [comment]
This issue is a duplicate of https://github.com/expressjs/expressjs.com/issues/695. Would be great to get some help on this!

**Comment by jcubic** [comment]
I've just noticed that HTTPS is working but there is a lack of redirect. Google Chrome opens HTTPS by default but if you click on the link from DuckDuckGo you get the website over HTTP:

https://duckduckgo.com/?q=express.js&ia=web

**Comment by jcubic** [comment]
Do you have the Enforce HTTPS option enabled for this repo?

https://docs.github.com/en/pages/getting-started-with-github-pages/securing-your-github-pages-site-with-https

If yes and it doesn't work, I would just contact GitHub support. No contributor can help you with this. Since settings require Admin rights to the repository.

**Comment by dougwilson** [comment]
Hi @jcubic girhub shows that setting is unavailable because there is a Cloudfront endpoint in front. This issue and other is us teying to get in contact with the appropriate folks to change it.

**Comment by emilsivervik** [comment]
Any updates on this?

**Comment by jcubic** [comment]
It looks like it works now. I believe that there is a new website that do work over https, and there is a redirect.

**Comment by UlisesGascon** [comment]
> It looks like it works now. I believe that there is a new website that do work over https, and there is a redirect.

Yep, seems like is working as expect now. So I will close the issue.

**Comment by bjohansebas** [comment]
The site still doesn't redirect to https when accessed via http. It might redirect some users to https due to a configuration in their browser (Firefox has that option). I've tried in Chrome, where I don't have any special settings, and it still doesn't redirect me to https

![image](https://github.com/user-attachments/assets/ec361040-ec74-4aac-8895-81ebb51c0f14)

**Comment by bjohansebas** [comment]
@crandmck Is there anything I can do to help resolve this?

**Comment by UlisesGascon** [comment]
Actually I just found the issue, it is in the pages configuration the "enforce policy" is not enabled. Seems like we need to do something on the DNS first. I will work on this with the Infra team (cc: @ryanaslett )

**Comment by ryanaslett** [comment]
Howdy, I just started to dig around in this, and turns out we don't have access to express.js' cloudflare account. I have in my notes that @UlisesGascon was going to ask about providing us access.

**Comment by bjohansebas** [comment]
Any updates on this?

**Comment by wesleytodd** [comment]
This is now complete. Thanks to @ryanaslett for the help getting these moved over.
